src/videos/src/videos-service/app.py

```diff
@@ -8,6 +8,8 @@
 
 patch_all()
 
+xray_recorder.begin_segment("Videos-init")
```

Comment:
This will cover the default mode of using the default streams but I don't think it deals with the case where channels are provided and background threads are started for each channel. For that, I think we need to either annotate the `stream` function to begin/end segments or add code within the `stream` function for this.